web/src/lib/services/api.ts

```diff
@@ -44,7 +45,9 @@ function fetchAPI(path, method, tokenValue, json, checkResponse) {
                 if (tokenValue) {
                     console.log("Error 401: Unauthorized. Deleting the token.");
                     token.set(null);
```

Comment:
This is the same as logout() above - why not move logout() to utils, and have it receive an (optiona) redirect parameter (default to "/", but here you would pass "/login"?